lib/health.js

```diff
@@ -16,55 +16,74 @@ function loadMainPackageJSON(attempts) {
 }
 
 var pjson = loadMainPackageJSON();
-
 var DEFAULT_PATH = '/ping';
 
-module.exports.ping = function pingMiddlewareWrapper(path) {
-  path = path || DEFAULT_PATH;
-  return function pingMiddleware(req, res, next) {
-    if (req.path === path) {
+/**
+ * Get system informaton
+ * @param  {Function} cb
+ */
+function info(cb) {
+  df(function (error, diskInfo) {
+    if (error) {
+      diskInfo = {msg: 'Error fetching disk info', error: error};
+    }
 
-      df(function (error, diskInfo) {
-        if (error) {
-          diskInfo = {msg: 'Error fetching disk info', error: error};
-        }
-        res.json({
-          timestamp: Date.now(),
-          uptime: process.uptime(),
+    var data = {
+      timestamp: Date.now(),
+      uptime: process.uptime(),
 
-          application: {
-            name: pjson.name,
-            version: pjson.version,
-            pid: process.pid,
-            title: process.title,
-            argv: process.argv,
-            versions: process.versions,
-            node_env: process.env.NODE_ENV
-          },
+      application: {
+        name: pjson.name,
+        version: pjson.version,
+        pid: process.pid,
+        title: process.title,
+        argv: process.argv,
+        versions: process.versions,
+        node_env: process.env.NODE_ENV
+      },
 
-          resources: {
-            memory: process.memoryUsage(),
-            loadavg: os.loadavg(),
-            cpu: os.cpus(),
-            disk: diskInfo,
-            nics: os.networkInterfaces()
-          },
+      resources: {
+        memory: process.memoryUsage(),
+        loadavg: os.loadavg(),
+        cpu: os.cpus(),
+        disk: diskInfo,
+        nics: os.networkInterfaces()
+      },
 
-          system: {
-            arch: process.arch,
-            platform: process.platform,
-            type: os.type(),
-            release: os.release(),
-            hostname: os.hostname(),
-            uptime: os.uptime(),
-            cores: os.cpus().length,
-            memory: os.totalmem()
-          }
-        });
-      });
+      system: {
+        arch: process.arch,
+        platform: process.platform,
+        type: os.type(),
+        release: os.release(),
+        hostname: os.hostname(),
+        uptime: os.uptime(),
+        cores: os.cpus().length,
+        memory: os.totalmem()
+      }
+    };
+
+    cb(data);
```

Comment:
I'm not against including promises but, in that case, I think we should keep the dual behaviour (callbacks or promises) to avoid breaking backwards compatibility.